devtron-labs/dashboard
# Enhancement - enabled the option to change chart version & values for external apps

# Description

Added the chart repo linking functionality for EA helm apps/charts under - [#1339](https://github.com/devtron-labs/devtron/issues/1339). However, after successfully linking there's no option to update the chart repo, version, or values.

These changes will enable the functionality to show chart repo, version, and values dropdown for external apps similar to Devtron charts even after linking the app to a helm chart repo.

Fixes # https://github.com/devtron-labs/devtron/issues/1390

## Type of change


# How Has This Been Tested?

Tests have been performed manually.

A. **EA - Without linking**
1. Go to AppDetails page 
2. Switch to the Values tab
3. Update required options/values
4. Click on update & modal will open
5. Click on update without linking
6. Observe

**Expected:** I should trigger the app update & after successful update redirect to AppDetails section

B. **EA - After linking**
1. Go to AppDetails page 
2. Switch to the Values tab
3. Update required options/values.
4. Click on update
5. Observe

**Expected:** I should trigger the app update & after successful update redirect to AppDetails section. Similar to existing Devtron chart flow.

C. **Devtron chart - API update**
1. Go to AppDetails page of the Devtron chart
2. Switch to the Values tab
3. Update required options/values.
4. Click on update
5. Observe

**Expected:** I should trigger the app update & after successful update redirect to AppDetails section. Similar to existing flow.

# Checklist:
